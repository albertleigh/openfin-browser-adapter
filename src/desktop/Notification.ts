import {BaseApiClass} from './Base';

export interface NotificationOptions {
    url?:string,
    message?:string,
}

export class Notification extends BaseApiClass{

    static staticMethods:string[]=[
        'getCurrent',
    ];

    static staticSyncMethods:string[]=[
    ];

    static instanceMethods:string[]=[
        'close',
        'sendMessage',
        'show',
        'sendMessageToApplication',
    ];

    static connection:any=null;


    constructor(options:NotificationOptions) {
        super();
        if(options && options.url){
            window.open(options.url,'_blank');
        }

        setTimeout(()=>{
            if (Notification.connection){
                Notification.connection.send({topic:'onNotificationMessage',message:options.message});
            }
        },500);
    }

    static async create(options:NotificationOptions){
        return new Notification(options);
    }
}