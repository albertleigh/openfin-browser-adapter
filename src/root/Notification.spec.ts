import { Notification } from './Notification';

describe('Desktop::Notification',()=>{

    let clazz:typeof Notification;

    beforeEach(()=>{
        clazz = Notification;
    });

    it('all unsupported static methods defined',()=>{
        expect(clazz.staticMethods).toMatchSnapshot();
    })

    it('all unsupported instance methods defined',()=>{
        expect(clazz.instanceMethods).toMatchSnapshot();
    })

});