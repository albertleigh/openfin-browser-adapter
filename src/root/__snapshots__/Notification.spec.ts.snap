// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop::Notification all unsupported instance methods defined 1`] = `
Array [
  "close",
  "sendMessage",
  "show",
  "sendMessageToApplication",
]
`;

exports[`Desktop::Notification all unsupported static methods defined 1`] = `
Array [
  "getCurrent",
]
`;
